main/resource-access.bs

```diff
@@ -75,6 +75,291 @@ Issue: Write Linked Data Platform section.
 Draft:
 A Solid data pod MUST conform to the LDP specification [[!LDP]].
 
+## Resource Metadata ## {#rm}
+
+### Background and Need ### {#rm-need}
+<em>This section is non-normative.</em>
+
+Resource Metadata introduces a mechanism for linking to metadata about resources
+in the Solid Ecosystem using HTTP Link headers. The metadata may serve as
+supplementary descriptions or, when supported by a Solid server implementation,
+may influence the behavior of the resources. A given resource might link to zero
+or more such related metadata representations. It is not meant to supplant the
+ability for the directly associated resource to self-describe.
+
+Examples of this mechanism in use include:
+
+<ul>
+  <li>A binary JPEG image with a Link header, pointing to a description that has
+  an RDF representation.</li>
+  <li>A container with a Link header, pointing to an ACL resource that governs
+  access controls for the contained resources.</li>
+  <li>A resource whose shape is constrained by a ShEx schema includes a Link
+  header that points to a metadata resource defining that constraint.</li>
+  <li>A resource with an associated set of configurable parameters includes a
+  Link header that points to a resource where those configurable parameters
+  reside.</li>
+</ul>
+
+### Metadata Discovery ###  {#rm-discovery}
+
+To discover the metadata resources directly associated with a given target
+resource, a client MUST issue a `HEAD` or `GET` request to the target resource
+URL and inspect the `Link` headers in the response, or make an HTTP `GET`
+request on the target resource URL to retrieve an RDF representation, and
+inspect the encoded RDF graph for relations of one or more metadata types. These
+may be carried out in either order, but if the first fails to result in
+discovering the metadata resource or the described resource, the second MUST be
+tried.
+
+Issue: It would seem that requiring that metadata MUST looked up through *both*
+Link headers and the RDF graph is redundant and invites synchronization
+problems. Suggest that the primary mechanism is through Link headers, and the
+client MAY or SHOULD inspect the RDF graph for any additional metadata.
```

Comment:
If I'm not mistaken, that Issue text is new; was added after the PR that got merged in data-interoperability-panel and before the PR here.

I've expressed implementation's capability at https://github.com/solid/data-interoperability-panel/pull/32#issuecomment-599977210 so the checks are not redundant without a reason. A client can stop after it checks one location (Link header or RDF body). There are cases where it may be necessary to check both, and that's obviously because it didn't discover the relation from the first check. There are also cases where only one needs to be checked and that depends on the target resource that the client is starting its discovery process from.

I feel that the background on why this was proposed is being overlooked: https://github.com/solid/data-interoperability-panel/issues/37#issuecomment-603110090 . I suppose we can come back to that...

If everything is left up to the server, it needs to advertise all of the "metadata" relations for each resource up front (and whether their targets have any content or not, but later expected to be updated by the client) at server-determined locations... because clearly the client has no way of adjusting that.

If the intention is to only link 1) Web documents and 2) have the serve control the relation, the Link header with rel is the only thing that's required for interop. There is no point in talking about discovery through RDF body unless there are some use cases that calls for it eg. requiring URIs with fragments, client controlling it. And in those cases, of course the Link header will not suffice; neither the relation can be expressed or the client have the means to express it.

The potential synchronisation issue is clear only when linking Web documents. If RDF body relations are allowed and subjects use URI fragments, synchronisation issue doesn't come up.

Reply:
Auxiliary resources such as access control, configuration and server-managed strictly operate on the level of HTTP resources, whereas others such as shapes and descriptions can alternatively work at the level of RDF graphs. For the former, only Link rel is required, and the latter can alternatively make use of the RDF body.